### PR TITLE
File-encryptor sample: fixed PKCS5 paddings for block aligned contents

### PR DESCRIPTION
**Pull request by soccerGB**
This sample does not impact this sample when running with input that's not block fully aligned.
But for other file input, it needs to pad an extra block to make it fully compliant with PKCS5 padding rules

### CODE REVIEW

File: samples/file-encryptor/host/host.cpp [hunk 1]
```diff
@@ -269,104 +277,67 @@ int encrypt_file(
...
-    // CIPHER_BLOCK_SIZE bytes. If the file size is not a multiple of
-    // CIPHER_BLOCK_SIZE-byte blocks, PKCS5 Padding was used to make it exactly
-    // a CIPHER_BLOCK_SIZE-byte block
-    if (leftover_bytes)
+    if (encrypt)
     {
-        unsigned char paddingtest[CIPHER_BLOCK_SIZE];
-        unsigned char paddingtest_ciphertext[CIPHER_BLOCK_SIZE];
+        // The CBC mode for AES assumes that we provide data in blocks of
+        // CIPHER_BLOCK_SIZE bytes. PKCS#5 padding was used in this sample. Pad
```
Comment (anakrish) [inline samples/file-encryptor/host/host.cpp]: Change past-tense to "This sample uses PKCS#5 padding."

File: samples/file-encryptor/host/host.cpp [hunk 2]
```diff
@@ -219,16 +219,24 @@ int encrypt_file(
...
     bytes_left = src_data_size;
+
     if (leftover_bytes)
     {
         bytes_left = src_data_size - leftover_bytes;
     }
     requested_read_size = DATA_BLOCK_SIZE;
     cout << "Host: start " << (encrypt ? "encrypting" : "decrypting") << endl;
+
+    // It loops through DATA_BLOCK_SIZE blocks one at a time and then followed
```
Comment (anakrish) [inline samples/file-encryptor/host/host.cpp]: one at a time and then follows by

File: samples/file-encryptor/host/host.cpp [hunk 3]
```diff
@@ -269,104 +277,67 @@ int encrypt_file(
...
+
+        bytes_read = fread(
+            plaintext_padding_buf,
+            sizeof(unsigned char),
+            leftover_bytes,
+            src_file);
+        if (bytes_read != leftover_bytes)
+            goto exit;
+
+        // PKCS5 Padding
```
Comment (anakrish) [inline samples/file-encryptor/host/host.cpp]: Question: In PKCS5 Padding, you just use the leftover_bytes as the padding value?

Comment (gupta-ak) [reply samples/file-encryptor/host/host.cpp]: You use CIPHER_BLOCK_SIZE - leftover_bytes as the padding. For example, if you have 11 bytes left, you need to pad 5 bytes to align it to the 16 byte block size. So, you'd pad it as five 5s. If your message is aligned to the block size, you add an additional block of filled with 16s. This ensures that you can always use the last byte of the padded plaintext to determine the padding.

Comment (gupta-ak) [reply samples/file-encryptor/host/host.cpp]: I think it's just for ease of decrypting. You don't need to store the plaintext length and you can use the last byte of the padded plaintext to determine the padding. There are probably other schemes that would do the same thing, but this method was standardized.

Comment (soccerGB) [reply samples/file-encryptor/host/host.cpp]: Akash is right, it's not for more secure reason, it's to make it possible to recognize  where the real end of the ciphertext is.

**Pull request merged**